File: alert/slack.py
import json
import urllib.request


class SlackNotifier:
    def __init__(self, webhook_url: str):
        self.webhook_url = webhook_url

    def send(self, msg: str) -> None:
        body = json.dumps({"text": msg}).encode()
        req = urllib.request.Request(
            self.webhook_url, data=body,
            headers={"Content-Type": "application/json"},
        )
        with urllib.request.urlopen(req, timeout=5) as resp:
            if resp.status != 200:
                raise RuntimeError(f"Slack returned {resp.status}")

File: alert/slack.go
package alert

import (
	"bytes"
	"encoding/json"
	"fmt"
	"net/http"
)

type SlackNotifier struct {
	webhookURL string
}

func (s *SlackNotifier) Send(msg string) error {
	body, _ := json.Marshal(map[string]string{"text": msg})
	resp, err := http.Post(s.webhookURL, "application/json", bytes.NewReader(body))
	if err != nil {
		return err
	}
	defer resp.Body.Close()
	if resp.StatusCode != http.StatusOK {
		return fmt.Errorf("slack returned %d", resp.StatusCode)
	}
	return nil
}
